### PR TITLE
Favorites List

### PR DESCRIPTION
**Pull request by Npikielny**
## Overview
Add a list of favorite foods to the account management page for users who are not logged in.

## Changes Made
- Added a table view to `BRBAccountViewController`
 - Removed a scroll view from `BRBAccountViewController` and moved the contents to a `UITableViewCell`
 - Fixed the format of favorite foods displayed for eateries

## Test Coverage

Manual testing

## Screenshots (delete if not applicable)

<details>

  <summary>Reformatted Favorite Foods</summary>

  
  IMG_7248

Food items are compacted as much as possible (transitioning from vertical alignment to horizontal alignment when possible), and when there are too many items to display, ellipses are added at the end.

</details>
<details>

  <summary>New Login Layout</summary>

  
  IMG_7247

</details>

### CODE REVIEW

File: Eatery/Controllers/BRB/BRBLoginViewController.swift [hunk 1]
```diff
@@ -58,28 +59,32 @@ class BRBLoginViewController: UIViewController {
...
-        let scrollView = UIScrollView(frame: .zero)
-        scrollView.alwaysBounceVertical = true
-        scrollView.keyboardDismissMode = .onDrag
-        view.addSubview(scrollView)
-        scrollView.snp.makeConstraints { make in
-            make.edges.equalToSuperview()
-        }
+        tableView = UITableView(frame: .zero, style: .grouped)
+        setupTableView()
+        tableView.translatesAutoresizingMaskIntoConstraints = false
```
Comment (whoiswillma) [inline Eatery/Controllers/BRB/BRBLoginViewController.swift]: Is this needed? If it is can you move it into `setupTableView`?

File: Eatery/Views/BRB/BRBLoginTableViewCell.swift [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+class BRBLoginTableViewCell: UITableViewCell {
+
+    override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+    }
+
+    func configure(stackView: UIStackView) {
+        stackView.removeFromSuperview()
+        stackView.snp.removeConstraints()
+        contentView.addSubview(stackView)
```
Comment (whoiswillma) [inline Eatery/Views/BRB/BRBLoginTableViewCell.swift]:
```suggestion
        stackView.removeFromSuperview()
        contentView.addSubview(stackView)
```
Removing from superview also removes constraints? Maybe different with snapkit

**Review by whoiswillma — APPROVED** [review]
Looks great overall. It's worth addressing (maybe this PR, maybe another) to update the favorites list when changes are made

File: Eatery/Controllers/BRB/BRBLoginViewController.swift [hunk 3]
```diff
@@ -58,28 +59,30 @@ class BRBLoginViewController: UIViewController {
...
             loginButton.isEnabled = !isLoading
+
+            tableView.layoutSubviews()
         }
     }
 
+    private var favoriteItems = Defaults[\.favoriteFoods]
+    private let favoriteId = "favorite"
+    private let loginId = "login"
+    var tableView: UITableView!
```
Comment (danielVebman) [inline Eatery/Controllers/BRB/BRBLoginViewController.swift:70]: Can be 

```suggestion
    var tableView = UITableView(frame: .zero, style: .grouped)
```

If you were starting from scratch I would do it this way, but if we use `!` in eatery then it's better to stay consistent.

File: Eatery/Controllers/BRB/BRBLoginViewController.swift [hunk 4]
```diff
@@ -95,6 +98,13 @@ class BRBLoginViewController: UIViewController {
         passwordTextField.text = nil
         netidTextField.resignFirstResponder()
         passwordTextField.resignFirstResponder()
+        DefaultsKeys.updateFoodLocations {
+            DispatchQueue.main.async {
+                self.tableView.reloadData()
+            }
+        }
+        favoriteItems = Defaults[\.favoriteFoods]
+        tableView.reloadData()
```
Comment (danielVebman) [inline Eatery/Controllers/BRB/BRBLoginViewController.swift]: It seems like you expect line 103 to execute before 106-7, but that might not happen. Would that be buggy?

File: Eatery/Controllers/BRB/BRBLoginViewController.swift [hunk 5]
```diff
@@ -58,28 +59,30 @@ class BRBLoginViewController: UIViewController {
...
             stackView.isUserInteractionEnabled = !isLoading
             loginButton.isEnabled = !isLoading
+
+            tableView.layoutSubviews()
         }
     }
 
+    private var favoriteItems = Defaults[\.favoriteFoods]
+    private let favoriteId = "favorite"
+    private let loginId = "login"
```
Comment (danielVebman) [inline Eatery/Controllers/BRB/BRBLoginViewController.swift]: These names are vague

File: Eatery/Controllers/BRB/BRBLoginViewController.swift [hunk 6]
```diff
@@ -259,3 +280,58 @@ extension BRBLoginViewController: UITextFieldDelegate {
...
+            return cell
+        }
+        let cell = tableView.dequeueReusableCell(withIdentifier: favoriteId) as! FavoriteTableViewCell
+        let name = favoriteItems[indexPath.item]
+        cell.configure(
+            name: name,
+            locations: Defaults[\.favoriteFoodLocations][name],
+            favorited: DefaultsKeys.isFavoriteFood(name)
+        )
+        return cell
```
Comment (danielVebman) [inline Eatery/Controllers/BRB/BRBLoginViewController.swift]: Personally I prefer

```
if ... {
    ...
    return cell
} else {
    ...
    return cell
}
```

As you wrote it, it reads like a guard, which suggests that `section == 0` is a rarer case than the `else`. Maybe I'm reading a lot into it, but that's my opinion.

**Review by danielVebman — APPROVED** [review]
Great stuff! I just don't really get the favorites UI. Why is it sometimes stacked and sometimes collapsed?

**Pull request merged**